v2/jdbc-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/JdbcToBigQueryReadWithPartitionsTest.java

```diff
@@ -0,0 +1,188 @@
+
+    JdbcToBigQuery.run(
+            options, JdbcToBigQuery.writeToBQTransform(options).withTestServices(bigQueryServices))
+        .waitUntilFinish();
+
+    assertThat(fakeDatasetService.getAllRows(PROJECT, DATASET, TABLE))
+        .isEqualTo(ImmutableList.of(new TableRow().set("BOOK_ID", 1).set("TITLE", "ABC").set("SELL_TIME", "2024-12-24 06:00:00")));
+  }
+
+  @Test
+  public void testE2EWithLongPartitionColumnTypeWithBounds() throws IOException, InterruptedException {
+    options.setPartitionColumn("BOOK_ID");
+    options.setPartitionColumnType("long");
+    options.setLowerBound("1");
+    options.setUpperBound("100");
+
+    JdbcToBigQuery.run(
+            options, JdbcToBigQuery.writeToBQTransform(options).withTestServices(bigQueryServices))
+        .waitUntilFinish();
+
+    assertThat(fakeDatasetService.getAllRows(PROJECT, DATASET, TABLE))
+        .isEqualTo(ImmutableList.of(new TableRow().set("BOOK_ID", 1).set("TITLE", "ABC").set("SELL_TIME", "2024-12-24 06:00:00")));
+  }
+
+  @Test
+  public void testE2EWithLongPartitionColumnTypeWithIncorrectBounds() throws IOException, InterruptedException {
+    options.setPartitionColumn("BOOK_ID");
+    options.setPartitionColumnType("long");
+    options.setLowerBound("1");
+    options.setUpperBound("2024-12-24");
+
+    assertThrows(NumberFormatException.class, 
+        () -> JdbcToBigQuery.run(options, JdbcToBigQuery.writeToBQTransform(options).withTestServices(bigQueryServices)).waitUntilFinish()); 
+  }
+
+  @Test
+  public void testE2EWithDateTimePartitionColumnType() throws IOException, InterruptedException {
+    options.setPartitionColumn("SELL_TIME");
+    options.setPartitionColumnType("datetime");
+
+    JdbcToBigQuery.run(
+            options, JdbcToBigQuery.writeToBQTransform(options).withTestServices(bigQueryServices))
+        .waitUntilFinish();
+
+    assertThat(fakeDatasetService.getAllRows(PROJECT, DATASET, TABLE))
+        .isEqualTo(ImmutableList.of(new TableRow().set("BOOK_ID", 1).set("TITLE", "ABC").set("SELL_TIME", "2024-12-24 06:00:00")));
+  }
+
+  @Test
+  public void testE2EWithDateTimePartitionColumnTypeWithBounds() throws IOException, InterruptedException {
+    options.setPartitionColumn("SELL_TIME");
+    options.setPartitionColumnType("datetime");
+    options.setLowerBound("2024-12-24 05:00:00");
+    options.setUpperBound("2024-12-24 06:00:00");
+
+    JdbcToBigQuery.run(
+            options, JdbcToBigQuery.writeToBQTransform(options).withTestServices(bigQueryServices))
+        .waitUntilFinish();
+
+    assertThat(fakeDatasetService.getAllRows(PROJECT, DATASET, TABLE))
+        .isEqualTo(ImmutableList.of(new TableRow().set("BOOK_ID", 1).set("TITLE", "ABC").set("SELL_TIME", "2024-12-24 06:00:00")));
+  }
+
+  public void testE2EWithDateTimePartitionColumnTypeWithIncorrectBounds() throws IOException, InterruptedException {
+    options.setPartitionColumn("SELL_TIME");
+    options.setPartitionColumnType("datetime");
```

Comment:
please also add spomewhere test for missing column type to validate backward compatibility and test for invalid type.